gensim/similarities/docsim.py

```diff
@@ -190,7 +190,10 @@ def __init__(self, output_prefix, corpus, num_features, num_best=None, chunksize
             # undocumented feature: set output_prefix=None to create the server in temp
             self.output_prefix = utils.randfname(prefix='simserver')
         else:
-            self.output_prefix = output_prefix
+            self.output_prefix = os.path.join(os.path.dirname(output_prefix), 'shard', '')   # '' to get the trailing slash
+            if not os.path.exists(self.output_prefix):
+                os.makedirs(self.output_prefix)
```

Comment:
yes, there should an except statement for when it fails to create ```shard``` directory. I'll update the pr asap, thanks.